pkg/auth/tokens/token_util.go

```diff
@@ -59,6 +60,15 @@ func IsExpired(token v3.Token) bool {
 	return durationElapsed.Seconds() >= ttlDuration.Seconds()
 }
 
+// IsIdleExpired checks if the idle session timeout was reached since last update.
+func IsIdleExpired(token v3.Token, lastTimeActivity metav1.Time) bool {
+	if token.ActivityLastSeenAt.IsZero() {
```

Comment:
More of a note. This line can cause a concern for someone reading the code if they don't know/remember that `metav1.Time.IsZero()` handles a `nil` receiver.
This definitely gave me a pause.

Reply:
I got your point.
I might add a comment about why we are not checking the `nil` value for `ActivityLastSeenAt`.
wdyt?
Something like, right above the `if token.ActivityLastSeenAt.IsZero() {` line :
```
// we are not checking that `token.ActivityLastSeenAt == nil`
// since `metav1.Time.IsZero()` handles a nil receiver already
```